File: gaia-sdk-javascript/src/graphql/request/type/SkillVersion.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * A skill version is a built version of a skill created by a SkillBuildJob
 */
export class SkillVersion extends Array<(_:VariableRegistry) => string> {
public _typeName = "SkillVersion";
    public skillRef = () => { 
        this.push(_ => "skillRef")
    };

    public version = () => { 
        this.push(_ => "version")
    };

    public created = () => { 
        this.push(_ => "created")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/SkillVersion.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class SkillVersion(list):
    """
    A skill version is a built version of a skill created by a SkillBuildJob
    """

    def skill_ref(self):
        self.append(lambda x: "skillRef")

    def version(self):
        self.append(lambda x: "version")

    def created(self):
        self.append(lambda x: "created")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
